ts_benchmark/utils/data_splitter.py

```diff
@@ -0,0 +1,91 @@
+    :return: A tuple containing the target DataFrame and the remaining DataFrame.
+    """
+    num_columns = df.shape[1]  # Total number of columns in the DataFrame
+
+    def parse_target_channel(
+        target_channel: Optional[List], num_columns: int
+    ) -> List[int]:
+        """
+        Parses the target_channel configuration to determine target column indices.
+
+        :param target_channel: Configuration for selecting target columns.
+        :param num_columns: Total number of columns in the DataFrame.
+        :return: A list of unique target column indices.
+        """
+        if target_channel is None:
+            return list(range(num_columns))  # Select all columns
+
+        target_columns = []
+        for item in target_channel:
+            if isinstance(item, int):
+                # Handle single integer index (supports negative indices)
+                actual_index = item if item >= 0 else num_columns + item
+                if 0 <= actual_index < num_columns:
+                    target_columns.append(actual_index)
+                else:
+                    raise IndexError(
+                        f"target_channel configuration error: Column index {item} is out of range (total columns: {num_columns})."
+                    )
+            elif isinstance(item, list) and len(item) == 2:
+                # Handle slice represented as a list, e.g., [2, 4] selects columns 2 and 3
+                start, end = item
+                start = start if start >= 0 else num_columns + start
+                end = end if end >= 0 else num_columns + end
+
+                if not (0 <= start < num_columns):
+                    raise IndexError(
+                        f"target_channel configuration error: Slice start index {item[0]} is out of range (total columns: {num_columns})."
+                    )
+                if not (0 <= end <= num_columns):
+                    raise IndexError(
+                        f"target_channel configuration error: Slice end index {item[1]} is out of range (total columns: {num_columns})."
+                    )
+                if start > end:
+                    raise ValueError(
+                        f"target_channel configuration error: Slice start index {start} is greater than end index {end}."
+                    )
+
+                # Add the range of indices to target_columns
+                slice_indices = list(range(start, end))
+                target_columns.extend(slice_indices)
+            else:
+                raise ValueError(
+                    f"target_channel configuration error: Invalid configuration item {item}."
+                )
+
+        # Remove duplicates while preserving order
+        target_columns_unique = list(dict.fromkeys(target_columns))
```

Comment:
dict keeps order only after python3.6, shall we consider the compatibility with older python versions?